### PR TITLE
#284 sign out api

### LINKED ISSUE #264: [Frontend] - Incorrect password still signs user in
**Describe the bug**
When you try to sign in with a correct email but incorrect password, it gives the error that the password is wrong, but it still signs you in. (After investigation, seems that it only occurs on accounts that already had a valid login)

**To Reproduce**
- Log into account page correct email but incorrect password
image
https://user-images.githubusercontent.com/25015601/105613435-7ab09a00-5d90-11eb-97eb-a7ec38842edc.mp4

**Expected behavior**
The expected behaviour should give the error but not sign in. It should leave you on the sign in page with the error.

### LINKED ISSUE #284: [Backend] - Implement sign out API
As a developer, I would like to destroy the existing cookie session when an user signs out from the website.

**Description** 
Currently, the sign out functionality does not remove the cookie session and this causes the website to continue acting like the user is logged in. This even causes the bug #264 so this story must fix this.
To complete this issue, you need to create an endpoint that clears cookie header of the incoming request. 

Stack Link: https://stackoverflow.com/questions/27978868/destroy-cookie-nodejs

**Tasks**
- [x] Task 1: Endpoint Creation
- [x] Task 2: Remove cookie header safely
- [x] Task 3: Create call from frontend to this API

**Acceptance Criteria**
API calls to the backend should not contain a filled cookies header.
Fixes #264

### PR DESCRIPTION
**Pull request by ratulkai93**
this PR will allow users to wipe out the cookie while log out, which was set during the login period, hence, improving security.

### CODE REVIEW

File: Server/src/controllers/authenticationController.ts [hunk 1]
```diff
@@ -49,6 +49,17 @@ export class AuthenticationController {
     }
   }
 
+  async createLogoutRequest(request: Request, response: Response, next: NextFunction) {
+    const tokenValue = request.cookies && request.cookies.token;
+    console.log('token: ', tokenValue)
+    response.cookie('token', 'tokenValue', { path: '/' });
+    response.clearCookie('token', { path: '/' });
+    response.end();
+    console.log('new token?: ', tokenValue)
```
Comment (ds161833) [inline Server/src/controllers/authenticationController.ts]: logs

File: Server/src/controllers/authenticationController.ts [hunk 2]
```diff
@@ -49,6 +49,17 @@ export class AuthenticationController {
     }
   }
 
+  async createLogoutRequest(request: Request, response: Response, next: NextFunction) {
+    const tokenValue = request.cookies && request.cookies.token;
+    console.log('token: ', tokenValue)
+    response.cookie('token', 'tokenValue', { path: '/' });
+    response.clearCookie('token', { path: '/' });
```
Comment (ds161833) [inline Server/src/controllers/authenticationController.ts:53]: so, the first one sets the token, the clearCookie probably clears it, do ;you need the`response.cookie('token', 'tokenValue', { path: '/' });`?

File: Server/upload/25a6488a2135bdeae7e26a8e9baac62f [hunk 3]
```diff
@@ -1,240 +0,0 @@
-{
-  "reference": {
-    "type": "book",
-    "publisher": "University of California Press",
-    "authors": [
-      {
-        "firstname": "Stanley",
-        "middlename": "P.",
```
Comment (ds161833) [inline Server/upload/25a6488a2135bdeae7e26a8e9baac62f]: restore this file it is used in tests 
image

File: Server/src/controllers/authenticationController.ts (hunk 2 above)
Comment (ratulkai93) [reply Server/src/controllers/authenticationController.ts:53]: according to expressJS docs, it says i have to? not entirely sure though
image

File: Server/src/controllers/authenticationController.ts (hunk 1 above)
Comment (ratulkai93) [reply Server/src/controllers/authenticationController.ts]: removed on the latest commit

File: Server/src/controllers/authenticationController.ts (hunk 2 above)
Comment (ds161833) [reply Server/src/controllers/authenticationController.ts:53]: oh i see, so you're saying it doesnt work without that line for you?

Comment (ratulkai93) [reply Server/src/controllers/authenticationController.ts:53]: i removed it and its still working. i guess you're right in this case

File: Server/upload/25a6488a2135bdeae7e26a8e9baac62f (hunk 3 above)
Comment (ratulkai93) [reply Server/upload/25a6488a2135bdeae7e26a8e9baac62f]: done

File: Server/src/controllers/authenticationController.ts [hunk 4]
```diff
@@ -51,7 +51,6 @@ export class AuthenticationController {
 
   async createLogoutRequest(request: Request, response: Response, next: NextFunction) {
     const tokenValue = request.cookies && request.cookies.token;
```
Comment (GD-Dheer) [inline Server/src/controllers/authenticationController.ts]: This line is now unused

**Review by GD-Dheer — APPROVED** [review]
Only need to remove that extra line. My other concerns have already been addressed! Nice work

File: Server/src/controllers/authenticationController.ts (hunk 4 above)
Comment (ratulkai93) [reply Server/src/controllers/authenticationController.ts]: removed on the latest commit

**Review by ds161833 — APPROVED** [review]

**Pull request merged**